### ISSUE TITLE
in 'shadow-dom' mode, class "bg-#d00" will be converted to “.bg-\\#dd0000 {...}”

### ISSUE THREAD

**Issue by NMTuan** [unocss/unocss#1943]
配置项
![image](https://user-images.githubusercontent.com/2149763/204994112-637858b6-1076-45d4-8fb4-f6f1c227d068.png)

vue文件
![image](https://user-images.githubusercontent.com/2149763/204994190-c5bf428d-4da4-4dec-ae4c-c2b63916eee2.png)

浏览器中输出的内容
![image](https://user-images.githubusercontent.com/2149763/204994394-81fda457-8c50-4f48-b8ce-aaee95adb93c.png)

依赖版本
![image](https://user-images.githubusercontent.com/2149763/204994572-3570b2ce-e11c-4137-8cef-0e2421aba27d.png)

输出的`.bg-\\#dd0000`选择器无法正确的被浏览器解析掉，正确的输出应该是`.bg-\#dd0000`

**Comment by sibbng** [comment]
This will be fixed in #1945. however, you are not opt-in to Vue's Web Components feature. Styles defined in your components will be hoisted to <head>. This will be problematic if you have multiple components like that. Because you will have duplicated CSS in your app. It's pointless to use `shadow-dom` that way. If you want to use UnoCSS with Web Compenents defined in Vue SFCs please refer to [this section in Vue Docs](https://vuejs.org/guide/extras/web-components.html#sfc-as-custom-element) for proper setup. Once you set up your components as described in the Vue Docs and when #1945 is released your styles will be isolated in shadow dom as you would expect.

**Comment by NMTuan** [comment]
> This will be fixed in #1945, however, you are not opt-in to Vue's Web Components feature. Styles defined in your components will be hoisted to `<head>`. This will be problematic if you have multiple components like that. Because you will have duplicated CSS in your app. It's pointless to use `shadow-dom` that way. If you want to use UnoCSS with Web Components defined in Vue SFCs please refer to [this section in Vue Docs](https://vuejs.org/guide/extras/web-components.html#sfc-as-custom-element) for proper setup. Once you set up your components as described in the Vue Docs and when #1945 is released your styles will be isolated in shadow dom as you would expect.

think you !
the project is here: https://github.com/NMTuan/base_vite_chrome_extension/tree/feature/20221122
I didn't think too much about it, it just works so far. 
in `content-scripts`, i will use the `shadow-dom` mode.
and in other files, i want use the default mode.
but the config only supports one mode.

**Comment by sibbng** [comment]
I haven't tested really but I believe you can initialize UnoCSS multiple times and pass the related file patterns in `include` option.

```ts
const presets = [
  presetAttributify(),
  presetUno(),
  presetIcons(),
  presetRemToPx()
]
// vite plugins
UnoCSS({ 
  mode: 'shadow-dom',
  include: ['src/content/*'], 
  presets 
}),
UnoCSS({ 
  include: ['src/options/*', 'src/popup/*'], 
  presets 
}),
```
